### PR TITLE
feat: Adds Retire Version target

### PR DESCRIPTION
**Pull request by TC-AVNP**
### Adds target to retire a specific Version of an API

`cli project  -t "$TOKEN" -p $PROJECTID retire version -v "v1" -i foo`

### Improves error handling by transversing the CMD tree and printing all nodes except the roots (stackit-cli and project).

Example for the usage of a nested target like **retire** + **version**:

`Failed to retire version! An error occurred with statuscode 404: not found`

### CODE REVIEW

File: cmd/stackit-api-manager/cmd/project.go [hunk 1]
```diff
@@ -68,6 +69,7 @@ func init() {
 
 	// retireCmd flags
 	retireCmd.Flags().StringVarP(&identifier, "identifier", "i", "", "API Identifier")
+	retireCmd.Flags().StringVarP(&apiVersion, "api-version", "v", "", "API Version")
```
Comment (bravenut) [inline cmd/stackit-api-manager/cmd/project.go:72]: This is redundant with the introduction of the new `retire version` target and it's not being used as far as I can see.

**Comment by bravenut** [comment]
I remember that we agreed on adding logic to require users to confirm on the `retire` option because they may not be aware that all API versions are retired when they do not use the `retire version` but `retire` target.

File: cmd/stackit-api-manager/cmd/project.go (hunk 1 above)
Comment (TC-AVNP) [reply cmd/stackit-api-manager/cmd/project.go:72]: Good catch, removing

**Review by bravenut — APPROVED** [review]
lgtm

File: cmd/stackit-api-manager/cmd/response_util.go [hunk 2]
```diff
@@ -147,7 +148,22 @@ func printErrorCLIResponse(cmd *cobra.Command, resp *http.Response) error {
 		return errRequestFailed
 	}
 
-	cmd.Printf("Failed to %s! An error occurred with statuscode %d: %s\n", cmd.Use, resp.StatusCode, errorMessage)
+	currentTarget := cmd
```
[the remark this answers is not in the document]
Comment (TC-AVNP) [reply cmd/stackit-api-manager/cmd/response_util.go:151]: Added

File: cmd/stackit-api-manager/cmd/response_util.go [hunk 3]
```diff
@@ -100,7 +101,7 @@ func printSuccessCLIResponseHumanReadable(cmd *cobra.Command, resp *http.Respons
 		}
 		cmd.Printf("API with identifier \"%s\" published successfully for project \"%s\" and stage \"%s\" (API-URL: \"%s\")\n", r.Identifier, r.ProjectID, r.Stage, r.APIURL)
 	case *retireResponse:
-		cmd.Printf("API with identifier \"%s\" retired successfully for project \"%s\"\n", r.Identifier, r.ProjectID)
+		cmd.Printf(r.HumanReadableMessage())
```
[the remark this answers is not in the document]
Comment (TC-AVNP) [reply cmd/stackit-api-manager/cmd/response_util.go:104]: Cannot, this command has two different messages
I would argue that all other should implement a HumanReadableMessage interface. 
A TODO has been created

File: cmd/stackit-api-manager/cmd/project_retire.go [hunk 4]
```diff
@@ -2,22 +2,54 @@ package cmd
...
 
 	"github.com/spf13/cobra"
 )
 
-const messageRetireSuccess = "API retired successfully"
+var (
+	version string
+)
+
+const messageRetireSuccess = "API %s retired successfully"
```
[the remark this answers is not in the document]
Comment (TC-AVNP) [reply cmd/stackit-api-manager/cmd/project_retire.go:17]: done

File: cmd/stackit-api-manager/cmd/project_retire.go [hunk 5]
```diff
@@ -2,22 +2,54 @@ package cmd
...
-	Identifier string `json:"identifier"`
-	ProjectID  string `json:"projectId"`
+	Identifier string  `json:"identifier"`
+	ProjectID  string  `json:"projectId"`
+	Version    *string `json:"version,omitempty"`
+}
+
+func (r retireResponse) HumanReadableMessage() string {
+	if r.Version != nil {
+		return fmt.Sprintf(messageHumanReadableRetireVersionSuccess, r.Identifier, *r.Version, r.ProjectID)
```
[the remark this answers is not in the document]
Comment (TC-AVNP) [reply cmd/stackit-api-manager/cmd/project_retire.go:31]: Kept it consistent, the other CMD are also const for their messages

**Review by Ricagraca — APPROVED** [review]
thanks for fixing the comments
lgtm

**Pull request merged**